mica/archive/aca_dark/dark_cal.py

```diff
@@ -104,7 +104,11 @@ def get_dark_cal_id(date, select='before'):
 
     :returns: dark cal id string (YYYYDOY)
     """
-    return _get_dark_cal_id_vector(date, select=select)
+    dark_id = _get_dark_cal_id_vector(date, select=select)
+    if not dark_id.shape:
+        # returning an instance of the type, not a numpy array
+        return dark_id.dtype.type(dark_id)
```

Comment:
I think that `dark_id.tolist()` is a more standard idiom to get back to a pure Python representation. In this case it will return a `str` object instead of a `numpy.str_` object. Even though they are basically compatible they are still different object types.

Reply:
I changed it to use `tolist()`.